arch/arm/core/aarch64/reset.S

```diff
@@ -18,132 +12,66 @@
 
 _ASM_FILE_PROLOGUE
 
-/*
- * Platform may do platform specific init at EL3.
- * The function implementation must preserve callee saved registers as per
- * AArch64 ABI PCS.
- */
-
-WTEXT(z_arch_el3_plat_init)
-SECTION_FUNC(TEXT,z_arch_el3_plat_init)
-ret
-
 /*
  * Reset vector
  *
  * Ran when the system comes out of reset. The processor is in thread mode with
  * privileged level. At this point, neither SP_EL0 nor SP_ELx point to a valid
  * area in SRAM.
  *
- * When these steps are completed, jump to z_arm64_prep_c(), which will finish
- * setting up the system for running C code.
  */
 
 GTEXT(__reset)
 SECTION_SUBSEC_FUNC(TEXT,_reset_section,__reset)
 
 GTEXT(__start)
 SECTION_SUBSEC_FUNC(TEXT,_reset_section,__start)
+	/* Initialize PSTATE (mask all exceptions, select SP_EL0). */
+	msr	DAIFSet, #0xf
```

Comment:
The idea is to unmask only when we are ready to handle a failure.

Reply:
Makes sense.

In my experience there's a bit of a tradeoff in early boot:

Ostriching is a valid approach and will allow surviving some failures (e.g. a single rogue write somewhere) and debugging without a debugger.

OTOH, failing loudly and immediately during early boot is often better if (and only if) you _do_ have a debugger. (It results in a state much closer to the failure, is the main reason.)

From here and elsewhere it seems like Zephyr's intent is to take the former approach.

(One potential variant option is to unmask early, but have an early-boot exception handler that just records minimal information about the failure in a standalone manner, and once the assert infrastructure is up you check to see if there were any exceptions recorded. This is useful for async aborts in particular because otherwise you have practically zero info - "something somewhere during boot had a failing write" - if you record information when you get the abort you at least know "something before this point had a failing write". May or may not be worth the complexity.)

(As an aside, I really wish that ARM didn't conflate "abort not at the specific instruction" and "abort without failing address".)